app/src/main/java/com/kickstarter/ui/activities/UpdateActivity.kt

```diff
@@ -61,105 +74,112 @@ class UpdateActivity : BaseActivity<UpdateViewModel.ViewModel?>(), KSWebView.Del
             .subscribe { projectUrl ->
                 openProjectExternally(projectUrl)
             }
+            .addToDisposable(disposables)
 
         viewModel.outputs.hasCommentsDeepLinks()
-            .filter { it == true }
-            .compose(bindToLifecycle())
-            .compose(observeForUI())
+            .filter { it }
+            .compose(observeForUIV2())
             .subscribe {
                 viewModel.inputs.goToCommentsActivity()
             }
+            .addToDisposable(disposables)
 
         viewModel.outputs.deepLinkToThreadActivity()
             .filter { it.second == true }
             .map { it.first }
-            .compose(bindToLifecycle())
-            .compose(observeForUI())
+            .compose(observeForUIV2())
             .subscribe {
                 viewModel.inputs.goToCommentsActivityToDeepLinkThreadActivity(it)
             }
+            .addToDisposable(disposables)
 
         viewModel.outputs.startRootCommentsActivity()
-            .compose(bindToLifecycle())
-            .compose(observeForUI())
+            .compose(observeForUIV2())
             .subscribe { update ->
                 startRootCommentsActivity(update)
             }
+            .addToDisposable(disposables)
 
         viewModel.outputs.startRootCommentsActivityToDeepLinkThreadActivity()
-            .compose(bindToLifecycle())
-            .compose(observeForUI())
+            .compose(observeForUIV2())
             .subscribe {
                 startRootCommentsActivityToDeepLinkThreadActivity(it)
             }
+            .addToDisposable(disposables)
 
         viewModel.outputs.startProjectActivity()
-            .compose(bindToLifecycle())
-            .compose(observeForUI())
+            .compose(observeForUIV2())
             .subscribe { uriAndRefTag ->
                 startProjectActivity(uriAndRefTag.first, uriAndRefTag.second)
             }
+            .addToDisposable(disposables)
 
         viewModel.outputs.startShareIntent()
-            .compose(bindToLifecycle())
-            .compose(observeForUI())
+            .compose(observeForUIV2())
             .subscribe { updateAndShareUrl ->
                 startShareIntent(updateAndShareUrl)
             }
+            .addToDisposable(disposables)
 
         viewModel.outputs.updateSequence()
-            .compose(bindToLifecycle())
-            .compose(observeForUI())
+            .compose(observeForUIV2())
             .subscribe { updateSequence ->
                 binding.updateActivityToolbar.updateToolbar.setTitle(ksString.format(resources.getString(R.string.social_update_number), "update_number", updateSequence))
             }
+            .addToDisposable(disposables)
 
         binding.updateActivityToolbar.shareIconButton.setOnClickListener {
             viewModel.inputs.shareIconButtonClicked()
         }
+
+        viewModel.provideIntent(intent)
+    }
+
+    override fun onNewIntent(intent: Intent?) {
```

Comment:
Noticed this new logic - can you explain what it does? Is this activity supposed to be a single instance?

Reply:
So before the activity was providing the intent to the viewmodel via a class function in BaseActivity, but since we are not using that for this anymore i just copied its functionality to this activity/viewmodel.  it just provides the intent to the viewmodel so it can grab its arguments

Reply:
From what I can see in documentation onNewIntent is for activities that are set to "singleTop", and it doesn't look like this activity is set to singleTop in the AndroidManifest, so I was curious if there is some other use of onNewIntent that we're using it for here?

Reply:
nop[e, and if thats the case I can just remove it, only realy did this because its how the previous activity was doing it and I didnt want to rock the boat too much